draft-peabody-dispatch-new-uuid-format-03.txt

```diff
@@ -321,1105 +322,694 @@ Internet-Draft               new-uuid-format               February 2022
-   timestamp_48.  The remaining 12 bits in timestamp_48 MUST be set to
-   0.
+   00000000-0000-0000-E800-000000000000
+   xxxxxxxx-xxxx-xxxx-NMxx-xxxxxxxxxxxx
 
-   By using implementation-specific timestamp sources it is not
-   guaranteed that devices outside of the application context are able
-   to extract and parse the timestamp from UUIDv8 without some pre-
-   existing knowledge of the source timestamp used by the UUIDv8
-   implementation.
+               Figure 9: UUIDv7 Version and Variant Examples
 
-4.5.2.  UUIDv8 Clock Sequence Usage
+   N:  Variant (1110 [E])
 
-   A clock sequence MUST be used with UUIDv8 as added sequencing
-   guarantees when multiple UUIDv8 will be created on the same clock
-   tick.  The amount of bits allocated to the clock sequence depends on
-   the precision of the timestamp source.  For example, a more accurate
-   timestamp source using nanosecond precision will require less clock
-   sequence bits than a timestamp source utilizing seconds for
-   precision.
+   M:  Version (0111 [7] or 1000 [8])
 
-   The UUIDv8 layout in Figure 6 generically defines two possible clock
-   sequence values that can leveraged:
+   While this specification makes use of the unused variant bits of 111
+   defined in [RFC4122], Section 4.1.1; the placement of the version
+   within the same octet as the variant leaves room for a future
+   variant.  As such the table below details the previous variant, new
+   variant and new variant that is reserved for future definition.
 
-   *  12 bit clock sequence using time_or_seq for use when the timestamp
-      is less than 48 bits which allows for 4095 UUIDs per clock tick.
-   *  8 bit clock sequence using seq_or_node when the timestamp uses
-      more than 48 bits which allows for 255 UUIDs per clock tick.
+    +------+------+------+------+------------------------------------+
+    | Msb0 | Msb1 | Msb2 | Msb3 | Description                        |
+    +------+------+------+------+------------------------------------+
+    | 1    | 0    | x    | x    | The variant specified by [RFC4122] |
+    +------+------+------+------+------------------------------------+
+    | 1    | 1    | 1    | 0    | The variant specified in this      |
+    |      |      |      |      | document.                          |
+    +------+------+------+------+------------------------------------+
+    | 1    | 1    | 1    | 1    | Reserved for future definition.    |
+    +------+------+------+------+------------------------------------+
 
-   An implementation MAY use both time_or_seq and seq_or_node for clock
-   sequencing however it is highly unlikely that 20 bits of clock
-   sequence are needed for a given clock tick.  Furthermore, more bits
-   from the node MAY be used for clock sequencing in the event that 8
-   bits is not sufficient.
+           Table 1: UUID Variant defined by this specification
 
+7.3.  Length (Variable, 128-bit Default)
```

Comment:
I have another question about encoding:

* how to encode a UUID to base-32?
* should I encode using 40-bit groups, from left to right, as in [RFC-4648, section 6](https://datatracker.ietf.org/doc/html/rfc4648#section-6)?
* or should I code using a generic algorithm that performs [modulo division](https://cs.stackexchange.com/questions/10318/the-math-behind-converting-from-any-base-to-any-base-no-cross-base)?

The RFC-4648 algorithm is much faster than the modulus division algorithm, but it produces a completely different result. To produce the same result with RFC-4648 algorithm, it is necessary to pad the low-order bits with zeros, so that the resulting bit sequence is multiple of 40.

There are actually 3 questions  :)

Reply:
As discussed by @bradleypeabody, I will open a discussion thread to hammer this out. In the meantime I will commit a version of Draft 03 that drops this topic until we can nail down the specifics.